### ISSUE TITLE
Switch to MD5-based cache buster in filename

### ISSUE THREAD

**Issue by rondevera** [mintdigital/asset_hat#18]
Currently, cache busting is managed by adding a bundle's latest Git SHA to the query string. Two problems:
- Some CDNs fail to recognize cache busters in query strings.
- This assumes that Git is being used for version control.

Change from `application.min.css?a1b2c3` to `application-a1b2c3.min.css` (like in Rails 3.1), and drop Git SHAs in favor of MD5 fingerprints.

**Comment by bob-p** [comment]
Is the plan to actually write the images out like this? If so did you think about just writing out the file paths with a cache buster and using nginx to rewrite?

**Comment by rondevera** [comment]
Yeah, the plan is to actually change the filenames. This is consistent with how Rails 3.1 handles the problem, and saves the developer from the hassle of configuring the web server (which isn't necessarily nginx).

**Comment by dwo** [comment]
+1 to this notion. I reckon not taking Git for granted is the right way to go here.

Our team has battled back and forth with several approaches and the query string tactic just hasn't been very reliable with our CDN.
